engine/proto/master copy.proto

```diff
@@ -0,0 +1,291 @@
+
+    rpc GetLeader(GetLeaderRequest) returns(GetLeaderResponse) {
+        option (google.api.http) = {
+            get: "/api/v1/leader"
+        };
+    }
+
+    rpc ResignLeader(ResignLeaderRequest) returns(google.protobuf.Empty) {
+        option (google.api.http) = {
+            post: "/api/v1/leader/resign"
+        };
+    }
+}
+
+service TaskScheduler {
+    rpc ScheduleTask(ScheduleTaskRequest) returns(ScheduleTaskResponse) {}
+
+    // ReportExecutorWorkload is called from executor to server master to report
+    // resource usage in executor.
+    rpc ReportExecutorWorkload(ExecWorkloadRequest) returns(ExecWorkloadResponse) {}
+}
+
+// Refer to: https://cloud.google.com/apis/design/standard_methods 
+service JobManager {
+    rpc CreateJob(CreateJobRequest) returns (Job){
+        option (google.api.http) = {
+            post: "/api/v1/jobs"
+            body: "job"
+        };
+    };
+
+    rpc GetJob(GetJobRequest) returns (Job){
+        option (google.api.http) = {
+            get: "/api/v1/jobs/{id=*}"
+        };
+    };
+
+    rpc ListJobs(ListJobsRequest) returns (ListJobsResponse){
+        option (google.api.http) = {
+            get: "/api/v1/jobs"
+        };
+    };
+
+    // NOTE: for the compatibility of existing openapi(ticdc) format,
+    // we use `/cancel` but not `:cancel`(google api suggested)
+    // refer to: https://cloud.google.com/apis/design/custom_methods
+    rpc CancelJob(CancelJobRequest) returns (Job){
+        option (google.api.http) = {
+            post: "/api/v1/jobs/{id=*}/cancel"
+        };
+    };
+
+    rpc DeleteJob(DeleteJobRequest) returns (google.protobuf.Empty){
+        option (google.api.http) = {
+            delete: "/api/v1/jobs/{id=*}"
+        };
+    }
+}
+
+message Selector {
+    enum Op {
+        OpUnknown = 0;
+        Eq = 1;
+        Neq = 2;
+        Regex = 3;
+    }
+
+    string label = 1;
+    string target = 2;
+    Op op = 3;
+}
+
+message HeartbeatRequest {
+    string executor_id = 1;
+    int32 resource_usage = 2;
+    int32 status = 3;
+
+    uint64 timestamp = 4;
+    uint64 ttl = 5;
+}
+
+message HeartbeatResponse {
+}
+
+message Executor {
+    string id = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+    // name is the readable name of the executor.
+    string name = 2;
+    string address = 3;
+    int64  capability = 4;
+    map<string, string> labels = 5;
+}
+
+message RegisterExecutorRequest {
+    Executor executor = 1;
+}
+
+message ListExecutorsRequest {
+}
+
+message ListExecutorsResponse {
+    repeated Executor executors = 1;
+}
+
+message Master {
+    string id = 1;
+    // name is the readable name of the master.
+    string name = 2;
+    string address = 3;
+    bool is_leader = 4;
+}
+
+message ListMastersRequest {
+}
+
+message ListMastersResponse {
+    repeated Master masters = 1;
+}
+
+
+message ScheduleTaskRequest {
+    string task_id = 1;
+    int64 cost = 2;
```

Comment:
What does cost mean?

Reply:
Can remove it, as well as the `cost based scheduler`